public/src/App.jsx

```diff
@@ -1,16 +1,29 @@
-import React, { useEffect, useState } from "react";
+import React, { useCallback, useEffect, useState } from "react";
 
 import Header from "./components/header";
 import { styles } from "./constants";
 import useAPI from "./hooks/useAPI";
 import NoteCreator from "./components/NoteCreator";
 import Notes from "./components/Notes";
 import { useUserContext } from "./contexts/useUserContext";
+import FieldCloud from "./components/FieldCloud";
+import { useFieldTransferContext } from "./contexts/useFieldTransferContext";
 
 const App = () => {
   const [fieldDefinitions, setFieldDefinitions] = useState();
+  const [newNote, setNewNote] = useState({ text: "", field_values: [] });
```

Comment:
While empty keys provides a sort of schema, this is js not ts, so maybe make empty object `{}` work and/or move to ts